### PR TITLE
[iDEAL] Web Redirect

### PR DESCRIPTION
**Pull request by mpetrenco**

### CODE REVIEW

File: Source/Extensions/NSError+Judo.m [hunk 1]
```diff
@@ -182,6 +182,14 @@ + (NSError *)judoResponseParseError {
                                                                 title:UnableToProcessRequestErrorTitle.localized]];
 }
 
++ (NSError *)judoMissingChecksumError {
+    return [NSError errorWithDomain:JudoErrorDomain
+                               code:JudoErrorGeneral_Error
+    userInfo:[self userDataDictWithDescription:@"error_checksum_missing_description".localized
```
Comment (stpopa) [inline Source/Extensions/NSError+Judo.m]: Let's do this in 2 steps to make it more readable,
1. userInfo = [self userDataDictWithDescription:@"error_checksum_missing_description".localized ....
2. return return [NSError errorWithDomain:JudoErrorDomain
                                code:JudoErrorGeneral_Error  userInfo:userInfo];

Comment (mpetrenco) [reply Source/Extensions/NSError+Judo.m]: Done.

File: Source/Managers/IDEALManager.m [hunk 2]
```diff
@@ -29,50 +29,84 @@
 #import "JPReference.h"
 #import "JPResponse.h"
 #import "JPTransactionData.h"
+#import "NSError+Judo.h"
 
 @interface IDEALManager()
```
Comment (stpopa) [inline Source/Managers/IDEALManager.m]: Let's rename it to `IDEALService`

Comment (stpopa) [reply Source/Managers/IDEALManager.m]: managers most of the time are singletons

File: Source/Model/JPTransactionData.h [hunk 3]
```diff
@@ -94,6 +94,11 @@ typedef NS_ENUM(NSUInteger, TransactionResult) {
  */
 @property (nonatomic, assign) TransactionType type;
 
+/**
+ *  The order identifier string used for the iDEAL bank transaction
+ */
+@property (nonatomic, assign) NSString *_Nullable orderId;
```
Comment (stpopa) [inline Source/Model/JPTransactionData.h]: let's have a separate object instead of adding all the properties to the `JPTransactionData`

Comment (mpetrenco) [reply Source/Model/JPTransactionData.h]: We might, although `JPTransactionData` already contains half of the response parameters, such as merchant reference, consumer reference, etc. and duplications might be a bad idea.

File: Source/Managers/IDEALService.m [hunk 4]
```diff
@@ -29,49 +29,84 @@
 #import "JPReference.h"
 #import "JPResponse.h"
 #import "JPTransactionData.h"
+#import "NSError+Judo.h"
+
```
Comment (stpopa) [inline Source/Managers/IDEALService.m]: extraline

**Pull request merged**